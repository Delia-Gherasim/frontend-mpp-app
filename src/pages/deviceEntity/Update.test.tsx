import {render} from '@testing-library/react';
import {describe, expect, it} from 'vitest';
import {IDevice} from '../../model/item.type';
import EditDevice from './Update';

const mockDevice: IDevice = {
    id: 1,
    category: 'Electronics',
    type: 'Type',
    brand: 'Brand',
    owner: 'Owner',
    accessories: true,
    warranty: false,
    date: new Date('2024-03-31'),
};

describe('EditDevice', () => {
    it('should match snapshot', () => {
        const {container} = render(
            <EditDevice
                data={mockDevice}
                onBackButtonHandle={() => {}}
                onSaveButtonHandler={() => {}}
            />,
        );

        expect(container).toMatchSnapshot();
    });
});
