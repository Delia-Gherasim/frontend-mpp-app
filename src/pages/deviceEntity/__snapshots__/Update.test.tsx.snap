// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditDevice > should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h3
      class="title"
    >
      Add New Device
    </h3>
    <form
      class="form-container"
    >
      <div>
        <label>
          Category:
        </label>
        <select>
          <option
            value=""
          >
            Select Category
          </option>
          <option
            value="Electronics"
          >
            Electronics
          </option>
          <option
            value="Appliances"
          >
            Appliances
          </option>
          <option
            value="Equipment"
          >
            Equipment
          </option>
        </select>
      </div>
      <div>
        <label>
          Type:
        </label>
        <input
          type="text"
          value="Type"
        />
      </div>
      <div>
        <label>
          Brand:
        </label>
        <input
          type="text"
          value="Brand"
        />
      </div>
      <div>
        <label>
          Owner:
        </label>
        <input
          type="text"
          value="Owner"
        />
      </div>
      <div>
        <label>
          Accessories:
        </label>
        <input
          checked=""
          type="checkbox"
        />
      </div>
      <div>
        <label>
          Warranty:
        </label>
        <input
          type="checkbox"
        />
      </div>
      <div>
        <label>
          Date:
        </label>
        <input
          type="date"
          value="2024-03-31"
        />
      </div>
      <div
        class="buttons"
      >
        <input
          type="button"
          value="Back"
        />
        <input
          type="submit"
          value="Save"
        />
      </div>
    </form>
  </div>
</div>
`;
